### ISSUE TITLE
`cuda-driver-dev` may need to expose the driver stub to `LD_LIBRARY_PATH`

### ISSUE THREAD

**Issue by leofang** [conda-forge/cuda-cudart-feedstock#5]
In pre-CUDA 12, it's expected that the `libcuda.so` stub is default loadable, but it seems this assumption no longer holds, and I have to [tweak it](https://github.com/conda-forge/cupy-feedstock/pull/199/commits/1f867dbb3191f655a2e772336745a3436d793286) so that the `import cupy` test (which loads the stub) works with CUDA 12.

**Comment by jakirkham** [comment]
Looking to add `cuda-cudart-dev_{{ target_platform }}` as a requirement of `cuda-nvcc-impl_{{ target_platform }}` in PR ( https://github.com/conda-forge/cuda-nvcc-feedstock/pull/11 ) to ensure it is present and near the compiler when it links it. Hoping that is enough

Would be a little worried about adding the stub libraries to `LD_LIBRARY_PATH` as it can cause issues in runtime environments (where maybe a compiler and these libraries are also needed for CuPy or Numba). Namely the stub libraries get picked up by the library loader instead of the actual libraries

So would suggest we try and see if we can solve this in the compiler package (maybe tweaking the search paths of the compiler itself)

**Comment by isuruf** [comment]
> In pre-CUDA 12, it's expected that the libcuda.so stub is default loadable, but it seems this assumption no longer holds

This is not true. Pre-CUDA 12, there is a `/usr/local/lib/libcuda.so.1` in the docker image we use which loaded by the dynamic linker. With CUDA 12, we use the regular docker image without that library.
